### PR TITLE
Add tests for #new and #create in CriteriaController when marks are r…

### PR DESCRIPTION
**Pull request by freyazjiner**
## Changes
*(Describe your changes here. Also describe the motivation for your changes: what problem do they solve, or how do they improve the application or codebase? If this pull request fixes an open issue, [use a keyword to link this pull request to the issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).)*

Adds tests for the “marks released” edge-case to both `#new` and `#create`. The approach is first create an assignment with released results, makes the corresponding request (`get_as` or `post_as`), and verifies that the controller returns a flash message with an error and an appropriate response code.

<details>
<summary>Screenshots of your changes (if applicable)</summary>

</details>

<details>
<summary>Associated <a href="https://github.com/MarkUsProject/Wiki">documentation repository</a> pull request (if applicable)</summary>

</details>

## Type of Change
*(Write an `X` or a brief description next to the type or types that best describe your changes.)*

| Type                                                                                    | Applies? |
|-----------------------------------------------------------------------------------------|----------|
| 🚨 *Breaking change* (fix or feature that would cause existing functionality to change) |          |
| ✨ *New feature* (non-breaking change that adds functionality)                          |          |
| 🐛 *Bug fix* (non-breaking change that fixes an issue)                                  |          |
| 🎨 *User interface change* (change to user interface; provide screenshots)              |          |
| ♻️ *Refactoring* (internal change to codebase, without changing functionality)          |          |
| 🚦 *Test update* (change that *only* adds or modifies tests)                            | X        |
| 📦 *Dependency update* (change that updates a dependency)                               |          |
| 🔧 *Internal* (change that *only* affects developers or continuous integration)         |          |

## Checklist
*(Complete each of the following items for your pull request. Indicate that you have completed an item by changing the `[ ]` into a `[x]` in the raw text, or by clicking on the checkbox in the rendered description on GitHub.)*

Before opening your pull request:

- [X] I have performed a self-review of my changes.
  - Check that all changed files included in this pull request are intentional changes.
  - Check that all changes are relevant to the purpose of this pull request, as described above.
- [ ] I have added tests for my changes, if applicable.
  - This is **required** for all bug fixes and new features.
- [X] I have updated the project documentation, if applicable.
  - This is **required** for new features.
- [X] If this is my first contribution, I have added myself to the list of contributors.

After opening your pull request:

- [x] I have updated the project Changelog (this is required for all changes).
- [X] I have verified that the pre-commit.ci checks have passed.
- [X] I have verified that the CI tests have passed.
- [X] I have reviewed the test coverage changes reported by Coveralls.
- [x] I have [requested a review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) from a project maintainer.

## Questions and Comments
*(Include any questions or comments you have regarding your changes.)*\
I split each controller action into two scenarios:
1. Marks not released – existing examples remain here.
2. Marks released – new examples that test the error-path.
Github tests all pass for now.

### CODE REVIEW

**Review by david-yz-liu — COMMENTED** [review]
@freyazjiner a good start. Here are a few general comments:

1. You deleted the pull request template that pre-populated the description field when making the pull request, which you should not do. You can find the pull request template under the `.github` folder; please use it to update your pull request description, and make sure to read through it and use the checklist to guide other things you need to do when making a pull request.
2. The tests should be added to the existing tests for `#new` and `#create`, rather than being created in a separate block. You can update just the context that involves "An authenticated and authorized instructor doing a GET". You'll need to figure out a good way to combine the setup required for these cases with the existing test cases.

File: Changelog.md [hunk 1]
```diff
@@ -12,6 +12,7 @@
 
 - Remove `activerecord-session_store` gem (#7517)
 - Upgrade to Rails 8 (#7504)
+- Tests added for #new and #create actions for CriteriaController (#7521)
```
Comment (david-yz-liu) [inline Changelog.md:15]: For consistency with other entries, phrase this as "Add tests for..." (i.e., starting with a verb in the imperative tense).

Also use backticks to surround all code names (`#new`, `#create`, `CriteriaController`).

File: spec/controllers/criteria_controller_spec.rb [hunk 2]
```diff
@@ -199,23 +199,49 @@
       end
 
       describe '#new' do
-        before do
-          get_as instructor,
-                 :new,
-                 params: { course_id: course.id, assignment_id: assignment.id },
-                 format: :js
-        end
+        context 'assignment marks are not released' do
```
Comment (david-yz-liu) [inline spec/controllers/criteria_controller_spec.rb:202]: For `context` blocks, the description should generally start with the word "when"

File: spec/controllers/criteria_controller_spec.rb [hunk 3]
```diff
@@ -199,23 +199,49 @@
...
+                   :new,
+                   params: { course_id: course.id, assignment_id: released_assignment.id },
+                   format: :js
+          end
+
+          it 'displays an error message' do
+            expect(flash[:error]).to have_message(I18n.t('criteria.errors.released_marks'))
+          end
+
+          it 'responds with appropriate status' do
```
Comment (david-yz-liu) [inline spec/controllers/criteria_controller_spec.rb:242]: Be more specific: responds with :bad_request status. Same comment below.

**Review by david-yz-liu — COMMENTED** [review]
@freyazjiner Good work! I just left a few minor comments.

File: Changelog.md [hunk 4]
```diff
@@ -6,12 +6,14 @@
 
 ### ✨ New features and improvements
 
+
```
[the remark this answers is not in the document]
Comment (freyazjiner) [reply Changelog.md:9]: sorry about that, done

[the remark this answers is not in the document]
Comment (david-yz-liu) [reply Changelog.md:9]: This line is still here (you may have reverted this change by accident). To be clear, I'm referring to the blank line that has been added; there should be just one blank line after the "New features and improvements" heading.

File: doc/markus-contributors.txt [hunk 5]
```diff
@@ -91,7 +92,7 @@ Horatiu Halmaghi
 Huiyu Sun
 Ian Smith
 Ibrahim Shahin
-Ido Ben Haim
+Igit revert HEADdo Ben Haim
```
Comment (david-yz-liu) [inline doc/markus-contributors.txt:95]: This line contains an error. Please make sure to check your work carefully when making commits, using `git diff` (and the GitHub pull request interface).

**Review by david-yz-liu — APPROVED** [review]
